app/src/main/java/com/topjohnwu/magisk/core/utils/ShellInit.kt

```diff
@@ -46,20 +41,22 @@ class ShellInit : Shell.Initializer() {
                 Info.noDataExec = !shell.newJob().add("$localBB sh -c \"$localBB true\"").exec().isSuccess
             }
 
+            val entry = if (shell.isRoot) "nsenter -t \$(pidof magiskd) -m sh" else "sh"
```

Comment:
I guess this will be broken if someone (apps) change their process name to "magiskd"?